src/NServiceBus.Core.Tests/Config/When_using_initialization.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace NServiceBus.Core.Tests.Config
+{
+    using System;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_using_initialization_with_non_default_ctor
+    {
+        [Test]
+        public void Should_throw_meaningful_exception()
```

Comment:
What do you think about moving this test to an acceptance test and also verify if for example the exception gets thrown to the user or if the endpoint starts anyway. I guess there should already be some acceptance tests about INeedInitialization (otherwise good chance to add some).